src/components/Secrets/utils/service-account-utils.ts

```diff
@@ -131,6 +131,60 @@ export const linkSecretToBuildServiceAccount = async (
     ? [...existingSecrets, { name: secret.metadata.name }]
     : [{ name: secret.metadata.name }];
 
+  // console.log("service acc",secret?.metadata?.name,component?.metadata?.name,imagePullSecretList,secretList)
+  return K8sQueryPatchResource({
+    model: ServiceAccountModel,
+    queryOptions: {
+      name: serviceAccountName,
+      ns: namespace,
+    },
+    patches: [
+      {
+        op: 'replace',
+        path: `/imagePullSecrets`,
+        value: imagePullSecretList,
+      },
+      {
+        op: 'replace',
+        path: `/secrets`,
+        value: secretList,
+      },
+    ],
+  });
+};
+
+export const linkSecretsToBuildServiceAccount = async (
+  secrets: SecretKind[],
+  component: ComponentKind,
+) => {
+  // When there is no secret/component or they are not in the same namespace, return
+  if (
+    !secrets ||
+    !component ||
+    secrets.find((secret) => secret.metadata.namespace !== component?.metadata?.namespace)
+  ) {
+    return;
+  }
+
+  const secretListString = secrets.map((item: SecretKind) => {
+    return { name: item?.metadata.name };
```

Comment:
we do not need to redefine the existing logic. 
 const results = await Promise.all(
    secrets.map(secret => linkSecretToBuildServiceAccount(secret, component))
  );
  return results